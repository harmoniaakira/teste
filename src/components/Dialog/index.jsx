import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Slide from '@material-ui/core/Slide';
import { Toolbar } from '@material-ui/core';
import ArrowBack from '@material-ui/icons/ArrowBackIos';
import Button from '../Button'
import {
  DialogFullScreenTitle,
  IconButton,
  AppBar
} from './Dialog.styles'

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const DialogSlide = ( 
  { 
    open, 
    title, 
    confirmButtonLabel, 
    cancelButtonLabel, 
    onConfirm,
    onCancel,
    handleClose,
    fullScreen,
    children
  } ) => {
  
  return (
    <Dialog
      open={open}
      TransitionComponent={Transition}
      keepMounted
      onClose={handleClose}
      fullScreen={fullScreen}
    >
      {
        fullScreen
          ? <AppBar sx={{ position: 'relative' }}>
              <Toolbar>
                <IconButton
                  edge="start"
                  color="inherit"
                  onClick={handleClose}
                  aria-label="close"
                >
                  <ArrowBack />
                </IconButton>
                <DialogFullScreenTitle>
                  {title}
                </DialogFullScreenTitle>
              </Toolbar>
            </AppBar>
          : <DialogTitle>{title}</DialogTitle>
      }
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {children}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        {
          cancelButtonLabel 
          ? <Button popup handleClick={onCancel}>{cancelButtonLabel}</Button>
          : ""
        }
        <Button popup={!fullScreen} handleClick={onConfirm}>{confirmButtonLabel}</Button>
      </DialogActions>
    </Dialog>
  );
}

DialogSlide.propTypes = {
	open: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  confirmButtonLabel: PropTypes.string.isRequired,
  cancelButtonLabel: PropTypes.string,
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  fullScreen: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

DialogSlide.defaultProps = {
	fullScreen: false,
  cancelButtonLabel: "",
  onCancel: null
}

export default memo(DialogSlide);
