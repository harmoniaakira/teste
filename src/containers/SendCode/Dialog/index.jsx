import { Dialog, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import { DialogActionsBetween, DialogBtn } from 'components/StyledUI/StyledUI.styles';
import { useSendCodeDialog, useSendCodeScreen } from 'contexts/SendCodeContext';
import React, { useState } from 'react';
import { SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN } from '../constants';
import { Flex } from '../SendCode.styles';

const SendCodeDialog = () => {
    const [phase, setPhase] = useState(0)
    const { isDialogSendCodeOpen, setIsDialogSendCodeOpen } = useSendCodeDialog()
    const { setScreen } = useSendCodeScreen()

    const handleClose = () => {
        setPhase(0)
        setIsDialogSendCodeOpen(false)
    }

    const Content = () => {
        switch (phase) {
            default:
                return "Certifique-se que seu cliente está em posse do celular informado. Deseja seguir com o código de validação?"

            case 1:
                return "Sem a validação do código de segurança, a solicitação estará sujeita a análise diferenciada da proposta. Deseja seguir sem o código de segurança?"

            case 2:
                return "Você poderá continuar a partir dessa etapa quando o cliente estiver com o celular em mãos. ;)"
        }
    }

    const handleAppraiseToken = () => {
        setPhase(0)
        setIsDialogSendCodeOpen(false)
        setScreen(SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN)
    }

    const RenderBtns = () => {
        switch (phase) {
            default:
                return <>
                    <DialogBtn onClick={() => setPhase(1)}>
                        Voltar
                    </DialogBtn>
                    <DialogBtn onClick={handleAppraiseToken}>
                        Continuar
                    </DialogBtn>
                </>
            case 1:
                return <>
                    <DialogBtn onClick={() => setPhase(2)}>
                        Voltar
                    </DialogBtn>
                    <DialogBtn onClick={handleClose} disabled>
                        Continuar
                    </DialogBtn>
                </>
            case 2:
                return <>
                    <Flex />
                    <DialogBtn onClick={handleClose}>
                        Fechar
                    </DialogBtn>
                </>
        }
    }

    return (
        <Dialog
            open={isDialogSendCodeOpen}
            onClose={handleClose}
        >
            <DialogTitle>Atenção</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <Content />
                </DialogContentText>
            </DialogContent>
            <DialogActionsBetween>
                <RenderBtns />
            </DialogActionsBetween>
        </Dialog>
    )
}

export default SendCodeDialog