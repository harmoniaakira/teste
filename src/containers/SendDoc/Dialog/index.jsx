import { DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import { DialogActionsBetween, DialogBtn, StyledDialog } from 'components/StyledUI/StyledUI.styles';
import { useSendDocDialog, useSendDocScreen } from 'contexts/SendDocContext';
import React from 'react';
import { SCREEN_SENDDOC_PRINT_FRONT } from '../constants';

const SendDocDialog = () => {

    const { isDialogSendDocOpen, setIsDialogSendDocOpen } = useSendDocDialog()
    const { setScreen } = useSendDocScreen()

    const handlePrintDoc = () => {
        setIsDialogSendDocOpen(false)
        setScreen(SCREEN_SENDDOC_PRINT_FRONT)
    }

    const handleClose = () => {
        setIsDialogSendDocOpen(false)
    }

    return (
        <StyledDialog
            open={isDialogSendDocOpen}
            onClose={handleClose}
        >
            <DialogTitle>Atenção</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    Será necessário aceitar o acesso à câmera do dispositivo para realizarmos a próxima etapa.
                </DialogContentText>
            </DialogContent>
            <DialogActionsBetween>
                <DialogBtn onClick={handleClose}>
                    Voltar
                </DialogBtn>
                <DialogBtn onClick={handlePrintDoc}>
                    Continuar
                </DialogBtn>
            </DialogActionsBetween>
        </StyledDialog>
    )
}

export default SendDocDialog