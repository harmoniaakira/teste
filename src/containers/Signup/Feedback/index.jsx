import React from 'react'
import AppHeader from 'components/AppHeader'
import Button from 'components/Button'
import { useStatusProposal } from 'contexts/SignupContext'
import PropTypes from 'prop-types'
import SignupAppraisal from './Appraisal'
import SignupError from './Error'
import SignupWarning from './Warning'
import SignupSuccess from './Success'
import { Container } from './SignupFeedback.styles'

const SignupFeedBack = ({ shouldRender }) => {
    const { statusProposal, setStatusProposal } = useStatusProposal()

    const RenderFeedback = () => {
        switch (statusProposal) {
            case 200:
                return <SignupSuccess />
            case 400:
                return <SignupError />
            case 422:
                return <SignupWarning />
            default:
                return (
                    <>
                        <SignupAppraisal />
                        <div style={{ display: 'flex', gap: '80px', margin: 20 }}>
                            <Button handleClick={() => setStatusProposal(200)}> Sucesso </Button>
                            <Button handleClick={() => setStatusProposal(400)}>  Erro </Button>
                            <Button handleClick={() => setStatusProposal(422)}>  AVISO </Button>
                        </div>
                    </>
                )
        }
    }

    return shouldRender ? (
        <>
            <AppHeader />
            <Container>
                <RenderFeedback />
            </Container>
        </>
    ) : null

}

SignupFeedBack.propTypes = {
    shouldRender: PropTypes.bool,
}

SignupFeedBack.defaultProps = {
    shouldRender: false
}
export default SignupFeedBack