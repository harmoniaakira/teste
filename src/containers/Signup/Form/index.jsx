import React, { useState } from 'react';
import { FormControl } from '@material-ui/core';
import MenuItem from '@material-ui/core/MenuItem';
import CheckBoxTwoToneIcon from '@material-ui/icons/CheckBoxTwoTone';
import Button from 'components/Button';
import { StyledCheckbox } from 'components/StyledCheckbox/StyledCheckbox.styles';
import { isValidDate } from 'utils/isValidDate';
import { isValidEmail } from 'utils/isValidEmail';
import { maskCpf } from 'utils/maskCpf';
import { maskDate } from 'utils/maskDate';
import { maskPhone } from 'utils/maskPhone';
import { isValidCpf } from 'utils/isValidCpf';
import { useCreditProposal } from 'contexts/SignupContext';
import { StyledArrowDown, StyledInput, StyledSelect, StyledSwitch, SytledInputLabel } from 'components/StyledUI/StyledUI.styles';
import { CheckCredit, CheckCreditConfirm, CheckCreditText, CheckPermissions, CheckPermissionsText, CheckPermissionsTitle, CheckWhatsapp, CheckWrapper, Container, Info, Row } from './FormSignup.styles';

const FormSignup = () => {
    const { setFillCreditProposal } = useCreditProposal()
    const [confirmCreditAnalysis, setConfirmCreditAnalysis] = useState(false)
    const [birthdayDate, setBirthdayDate] = useState("")
    const [phoneNumber, setPhoneNumber] = useState("")
    const [email, setEmail] = useState("")
    const [cpf, setCpf] = useState(null)

    const handleInputEvent = (event, maxLength, currState) => {
        const { value } = event.target
        if (value.length > maxLength) return currState
        return value
    }

    const handleBirthdayDateChange = (event) => {
        const value = handleInputEvent(event, 10, birthdayDate)
        setBirthdayDate(value)
    }

    const handlePhoneNumberChange = (event) => {
        const value = handleInputEvent(event, 15, phoneNumber)
        setPhoneNumber(value)
    }

    const handleCpfChange = (event) => {
        const value = handleInputEvent(event, 14, cpf)
        setCpf(value)
    }

    const handleCpfErrors = () => {
        if (isValidCpf(cpf) || cpf === null) {
            return ""
        }
        if (!isValidCpf(cpf)) {
            if (cpf.trim().length === 0) return "CPF não informado"
            return "CPF Inválido"
        }
        return ""
    }

    const handleRequestProposal = () => {
        // TODO: validar com api
        setFillCreditProposal(true)
    }

    return (
        <Container>
            <StyledInput
                error={!isValidCpf(cpf)}
                helperText={handleCpfErrors()}
                value={maskCpf(cpf)}
                onChange={handleCpfChange}
                label="CPF"
                placeholder="000.000.000-00"
                $placeholderghost
                fullWidth
                InputLabelProps={{
                    shrink: true,
                }}
            />
            <StyledInput
                label="Data de nascimento igual à identidade"
                fullWidth
                value={maskDate(birthdayDate)}
                onChange={handleBirthdayDateChange}
                error={isValidDate(birthdayDate).length > 0}
                helperText={isValidDate(birthdayDate)}
            />

            <StyledInput
                label="Informe um número de celular válido"
                fullWidth
                onChange={handlePhoneNumberChange}
                value={maskPhone(phoneNumber)}
            />

            <CheckWhatsapp>
                Este número de celular possui WhatsApp?
                <StyledSwitch />
            </CheckWhatsapp>

            <StyledInput
                label="Coloque aqui o e-mail principal do cliente"
                fullWidth
                error={!isValidEmail(email)}
                helperText={`${!isValidEmail(email) ? "E-mail inválido." : ""}`}
                onBlur={(e) => setEmail(e.target.value)}
            />
            <FormControl>
                <SytledInputLabel id="select-special-needs">Possui necessidade especial? </SytledInputLabel>
                <StyledSelect
                    labelId="select-special-needs"
                    IconComponent={StyledArrowDown}
                >
                    <MenuItem value={10}>Não</MenuItem>
                    <MenuItem value={20}>Dificuldade de visão</MenuItem>
                    <MenuItem value={30}>Dificuldade de audição e/ou fala</MenuItem>
                    <MenuItem value={40}>Limitações motoras</MenuItem>
                </StyledSelect>
            </FormControl>

            <CheckWrapper>
                <CheckPermissions>
                    <CheckPermissionsTitle> Aceites e permissões </CheckPermissionsTitle>
                    <CheckPermissionsText>
                        Para continuar, garanta que o cliente está ciente das opções abaixo.
                    </CheckPermissionsText>
                </CheckPermissions>


                <CheckCredit>
                    <CheckCreditConfirm>
                        Consulta sistemas de analise crédito
                        <StyledCheckbox
                            $colorone="#FFFFFF"
                            $colortwo="#757575"
                            color="primary"
                            checkedIcon={<CheckBoxTwoToneIcon />}
                            value={confirmCreditAnalysis}
                            onChange={() => setConfirmCreditAnalysis(!confirmCreditAnalysis)}
                        />
                    </CheckCreditConfirm>

                    <CheckCreditText>
                        Informe o cliente que ao fazer a consulta de dados no Banco Central,
                        sua condição de crédito poderá melhorar e isso facilitará na análise da proposta.
                    </CheckCreditText>
                </CheckCredit>
            </CheckWrapper>

            <Row>
                <Info />

                <Button disabled={!confirmCreditAnalysis} handleClick={handleRequestProposal}>
                    Continuar
                </Button>
            </Row>
        </Container>
    )
}

export default FormSignup