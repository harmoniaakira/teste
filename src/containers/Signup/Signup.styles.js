import styled from 'styled-components'
import ArrowBack from '@material-ui/icons/ArrowBackIos';

export const SignupHeader = styled.div`
  display: flex;
  align-items: center;
  gap: 30px;
  margin-left: -35px;
`

export const SignupHeaderArrow = styled(ArrowBack)`
  color: #8E8E93;
  cursor: pointer;
`

export const SignupHeaderText = styled.div`
  font-weight: bold;
  font-size: 20px;
  line-height: 150%;
  color: #636366;
  max-width: 280px;
`

export const WrapperFormSignup = styled.div`
  display: flex;
  flex-direction: column;
  @media (min-width: 1440px) {
    margin-top: 60px;
    max-width: 600px;
  }
  ${({ $hide }) => $hide && 'display: none;'}
`