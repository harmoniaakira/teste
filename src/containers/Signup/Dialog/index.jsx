import React from 'react'
import { useToggleDialog } from 'contexts/StepperContext'
import Dialog from 'components/Dialog'

const DialogSignup = () => {
    const { isDialogOpen, setIsDialogOpen } = useToggleDialog()

    const handleClose = () => {
        setIsDialogOpen(false)
    }

    return (
        <Dialog
            open={isDialogOpen}
            title="Preenchimento de dados"
            confirmButtonLabel="voltar"
            onConfirm={handleClose}
            handleClose={handleClose}
            fullScreen
        >
            <strong>CPF</strong><br />
            Insira somente os 11 dígitos do CPF do cliente. Ex.: 000.000.000-00
            <br /><br />
            <strong>Data de Nascimento</strong><br />
            <span>Preencha a Data de Nascimento do cliente, igual ao do documento de identificação. Ex.: 00/00/0000</span>
            <br /><br />
            <strong>Número de Celular válido</strong><br />
            <span>É necessário informar um número de celular válido. Ex.: (00) 00000-0000</span>
            <br /><br />
            <strong>E-mail principal válido</strong><br />
            <span>Importante informar o e-mail principal que é utilizado e também que seja válido. Ex.: contatodamaria@gmail.com</span>
        </Dialog>
    )
}

export default DialogSignup