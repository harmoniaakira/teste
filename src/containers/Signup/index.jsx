import { MobileView } from 'components/Layout/Layout.styles'
import Stepper from 'components/Stepper'
import Web from 'components/Web'
import { useCreditProposal } from 'contexts/SignupContext'
import React from 'react'
import { useHistory } from 'react-router'
import DialogSignup from './Dialog'
import SignupFeedBack from './Feedback'
import FormSignup from './Form'
import { SignupHeader, SignupHeaderArrow, SignupHeaderText, WrapperFormSignup } from './Signup.styles'



const Signup = () => {
  const history = useHistory()
  
  const { fillCreditProposal } = useCreditProposal()

  const handleClick = () => {
    history.push('/onboarding-pf-frontend/geolocalizacao');
  }

  return (
    <>
      <MobileView>
        <SignupFeedBack shouldRender={fillCreditProposal} />
        <WrapperFormSignup $hide={fillCreditProposal}>
          <Stepper infoButton />
          <FormSignup />
        </WrapperFormSignup>
      </MobileView>

      <Web>
        <SignupFeedBack shouldRender={fillCreditProposal} />
        <WrapperFormSignup $hide={fillCreditProposal}>
          <SignupHeader>
            <SignupHeaderArrow onClick={handleClick} />
            <SignupHeaderText>
              Preencha os campos com as informações do cliente
            </SignupHeaderText>
          </SignupHeader>
          <FormSignup />
        </WrapperFormSignup>
      </Web>

      <DialogSignup />
    </>
  )
}

export default Signup;
