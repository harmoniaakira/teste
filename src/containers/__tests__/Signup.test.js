import Dialog from 'components/Dialog';
import Signup from 'containers/Signup';
import DialogSignup from 'containers/Signup/Dialog';
import { shallow } from 'enzyme';
import React from 'react';

describe('Signup', () => {
  const signup = shallow(<Signup />);

  it('should match snapshot', () => {
    // signup.find(DialogSlide).props().handleClose();
    expect(signup).toMatchSnapshot()
  })
})

describe('Signup/Dialog', () => {
  const dialogSignup = shallow(<DialogSignup />);

  it('should render', () => {
    dialogSignup.find(Dialog).props().handleClose()
    expect(dialogSignup).toHaveLength(1);
  })
})