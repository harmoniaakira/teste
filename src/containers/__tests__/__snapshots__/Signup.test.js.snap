// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <SignupFeedBack
      shouldRender={false}
    />
    <styled.div
      $hide={false}
    >
      <Memo(Stepper)
        infoButton={true}
      />
      <FormSignup />
    </styled.div>
  </styled.div>
  <Web>
    <SignupFeedBack
      shouldRender={false}
    />
    <styled.div
      $hide={false}
    >
      <styled.div>
        <Styled(Component)
          onClick={[Function]}
        />
        <styled.div>
          Preencha os campos com as informações do cliente
        </styled.div>
      </styled.div>
      <FormSignup />
    </styled.div>
  </Web>
  <DialogSignup />
</Fragment>
`;
